waspc/data/Generator/templates/server/src/auth/providers/email/requestPasswordReset.ts

```diff
@@ -24,37 +27,48 @@ export function getRequestPasswordResetRoute({
     return async function requestPasswordReset(
         req: Request<{ email: string; }>,
         res: Response,
-    ): Promise<Response<{ success: true } | { success: false; message: string }>> {
-        const args = req.body || {};
+    ): Promise<Response<{ success: true }> | void> {
+        const args = req.body ?? {};
         ensureValidEmail(args);
 
-        args.email = args.email.toLowerCase();
+        const authIdentity = await findAuthIdentity(
+            createProviderId("email", args.email),
+        );
 
-        const user = await findUserBy({ email: args.email });
-    
         // User not found or not verified - don't leak information
-        if (!user || !user.isEmailVerified) {
+        if (!authIdentity) {
             await doFakeWork();
             return res.json({ success: true });
         }
 
-        if (!isEmailResendAllowed(user, 'passwordResetSentAt')) {
-            return res.status(400).json({ success: false, message: "Please wait a minute before trying again." });
+        const providerData = deserializeAndSanitizeProviderData<'email'>(authIdentity.providerData);
+        if (!providerData.isEmailVerified) {
+            await doFakeWork();
+            return res.json({ success: true });
         }
-    
-        const passwordResetLink = await createPasswordResetLink(user, clientRoute);
+
+        if (!isEmailResendAllowed(providerData, 'passwordResetSentAt')) {
+            throw new HttpError(400, "Please wait a minute before trying again.");
```

Comment:
![Screenshot 2023-12-29 at 13 15 26](https://github.com/wasp-lang/wasp/assets/2223680/e4db092d-381a-4439-aff9-2f499a76ad74)

I've seen this error and then I decided to add the `void` return value since I assumed as well it's connected to the `throw` statements.

But was a mistake on my part since `res.json` returns a response in Express **but** you don't have to `return res.json` in order for it to work properly. 

Why is this important? Because in one of the functions, I didn't have a `return` for each of the `res.json` calls which confused Typescript and then it demanded I modify the return type. It worked for Express, but Typescript wasn't happy.

After returning after each `res.json` the `void` return type can be safely removed.

I'll modify the code to reflect this new understanding 👍

Reply:
I've made it so now 👍